test/e2e/artifact_test.go

```diff
@@ -196,4 +180,155 @@ var _ = Describe("Podman artifact", func() {
 
 	})
+
+	It("podman artifact simple add --append", func() {
+		artifact1File, err := createArtifactFile(1024)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact2File, err := createArtifactFile(2048)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact3File, err := createArtifactFile(4192)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact1Name := "localhost/test/artifact1"
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", artifact1Name, artifact1File}...)
+
+		_ = podmanTest.InspectArtifact(artifact1Name)
+
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", "--append", artifact1Name, artifact2File}...)
+
+		a := podmanTest.InspectArtifact(artifact1Name)
+
+		Expect(a.Manifest.Layers).To(HaveLen(2))
+
+		annotation1 := "color=blue"
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", "--append", "--annotation", annotation1, artifact1Name, artifact3File}...)
+
+		a = podmanTest.InspectArtifact(artifact1Name)
+		Expect(a.Name).To(Equal(artifact1Name))
+		Expect(a.Manifest.Layers).To(HaveLen(3))
+
+		for _, l := range a.Manifest.Layers {
+			if l.Annotations["org.opencontainers.image.title"] == filepath.Base(artifact3File) {
+				Expect(l.Annotations["color"]).To(Equal("blue"))
+			} else {
+				Expect(l.Annotations).To(HaveLen(1))
+			}
+		}
+
+		listSession := podmanTest.PodmanExitCleanly([]string{"artifact", "ls"}...)
+		Expect(listSession.OutputToStringArray()).To(HaveLen(2))
+	})
+
+	It("podman artifact add --append multiple", func() {
+		artifact1File, err := createArtifactFile(1024)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact2File, err := createArtifactFile(2048)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact3File, err := createArtifactFile(4192)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact1Name := "localhost/test/artifact1"
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", artifact1Name, artifact1File}...)
+
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", "--append", artifact1Name, artifact2File, artifact3File}...)
+
+		a := podmanTest.InspectArtifact(artifact1Name)
+
+		Expect(a.Manifest.Layers).To(HaveLen(3))
+	})
+
+	It("podman artifact add --append modified file", func() {
+		artifact1File, err := createArtifactFile(1024)
+		Expect(err).ToNot(HaveOccurred())
+
+		artifact1Name := "localhost/test/artifact1"
+		podmanTest.PodmanExitCleanly([]string{"artifact", "add", artifact1Name, artifact1File}...)
+
+		f, err := os.OpenFile(artifact1File, os.O_APPEND|os.O_WRONLY, 0644)
+		Expect(err).ToNot(HaveOccurred())
+		_, err = f.WriteString("This is modification.")
+		Expect(err).ToNot(HaveOccurred())
+		f.Close()
+
+		appendFail := podmanTest.Podman([]string{"artifact", "add", "--append", artifact1Name, artifact1File})
+		appendFail.WaitWithDefaultTimeout()
+		Expect(appendFail).Should(Exit(125))
+		Expect(appendFail.ErrorToString()).Should(Equal(fmt.Sprintf("Error: file: %s already exist in artifact", filepath.Base(artifact1File))))
```

Comment:
Not blocking but generally error checks should be combined with `ExitWithError(code, "msg")` so it only takes one line and so it is not forgotten to check the error message.

same in the other places below